src/midcom/datamanager/extension/type/privilege.php

```diff
@@ -0,0 +1,156 @@
+            return $return_options;
+        };
+        $resolver->setDefaults(array(
+            'choices' => $map_privilege,
+            'choices_as_values' => true,
+            'expanded' => true,
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        parent::buildView($view, $form, $options);
+        $view->vars['privilege'] = $this;
+        $view->vars['type_conf'] = $options['type_config'];
+    }
+
+    public function get_effective_value(array $options,$object = null)
+    {
+        if (!$object)
+        {
+            $defaults = midcom::get()->auth->acl->get_default_privileges();
+            return(($defaults[$options['privilege_name']] === MIDCOM_PRIVILEGE_ALLOW));
+        }
+        if ($options['assignee']== 'SELF') {
+            if ($object instanceof midcom_db_group) {
+                //There's no sane way to query group privileges in auth right now, so we only return defaults
+                $defaults = midcom::get()->auth->acl->get_default_privileges();
+                return(($defaults[$options['privilege_name']] === MIDCOM_PRIVILEGE_ALLOW));
+            }
+            return(midcom::get()->auth->can_user_do($options['privilege_name'],
+                    new midcom_core_user($object->__object->id),$options['classname']));
+        }
+        if ($principal = midcom::get()->auth->get_assignee($options['assignee'])) {
+            return $object->can_do($options['privilege_name'], $principal);
+        }
+        return $object->can_do($options['privilege_name'], $options['assignee']);
+    }
+
+    public function render_choices(array $options,$object = null)
+    {
+            $l10n = midcom::get()->i18n->get_l10n('midcom.datamanager');
+            $inherit = $this->get_effective_value($options,$object);
+            $allow = 'widget privilege: allow';
+            $deny = 'widget privilege: deny';
+
+            if ($inherit === true) {
+                $base = $l10n->get('widget privilege: inherit %s');
+                $allow = $l10n->get($allow);
+                $d = sprintf($base,$allow);
+            } elseif ($inherit === false) {
+                $base = $l10n->get('widget privilege: inherit %s');
+                $deny = $l10n->get($deny);
+                $d = sprintf($base,$deny);
+            }
+            return $d;
+    }
+
+    public function search_for_object($object)
+    {
+        $object_found = false;
+        $help_obj = $object;
+
+        while (!$object_found)
+        {
+            if ($help_obj instanceof dbacontainer) {
+                $object_found = true;
+                return $help_obj->get_value();
+            } else {
+                if (!empty($help_obj->parent)) {
+                    $help_obj = $help_obj->parent->vars['data'];
+                } else {
+                    return null;
+                }
+            }
```

Comment:
The logic here can be simplified: Since you return when something is founds (or there's nothing else to look at), `$object_found` isn't needed (i.e. it will never be `true` in the `while` check)